### PR TITLE
ci: Split VSTS builds into build and test

### PR DESCRIPTION
**Pull request by jkleinsc**
##### Description of Change
This is a similar change to #14640.
This PR breaks out building and testing of Electron into separate jobs on VSTS.  This will help us to make sure that VSTS tests don't run more than 30 minutes and should give us better control over our jobs on VSTS

cc @alexeykuzmin 

##### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes

Notes: no-notes

### CODE REVIEW

File: vsts.yml [hunk 1]
```diff
@@ -78,24 +79,21 @@ jobs:
     inputs:
       PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/args.gn'
       ArtifactName: Default
+    condition: and(succeeded(), eq(variables['RUN_TESTS'], '1'))
```
Comment (nornagon) [inline vsts.yml]: why only publish these artifacts if the tests succeeded?

File: vsts.yml [hunk 2]
```diff
@@ -51,39 +52,89 @@ jobs:
...
 
+  - bash: |
+      cd src
+      ninja -C out/Default electron:electron_dist_zip
+    displayName: Build dist zip
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish Electron.app
+    inputs:
+      PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/Electron.app'
```
Comment (nornagon) [inline vsts.yml]: why publish this as well as the zip?

File: vsts.yml [hunk 3]
```diff
@@ -51,39 +52,89 @@ jobs:
...
+    displayName: Publish gn args
+    inputs:
+      PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/args.gn'
+      ArtifactName: Default
+    condition: and(succeeded(), eq(variables['RUN_TESTS'], '1'))
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish build.ninja
+    inputs:
+      PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/build.ninja'
```
Comment (nornagon) [inline vsts.yml]: what is this needed for?

File: vsts.yml [hunk 4]
```diff
@@ -51,39 +52,89 @@ jobs:
...
+    displayName: Publish Electron.app
+    inputs:
+      PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/Electron.app'
+      ArtifactName: Default
+
+  - task: PublishBuildArtifacts@1
+    displayName: Publish gn args
+    inputs:
+      PathtoPublish: '$(System.DefaultWorkingDirectory)/src/out/Default/args.gn'
+      ArtifactName: Default
```
Comment (nornagon) [inline vsts.yml]: do these artifact names collide?

**Comment by jkleinsc** [comment]
Closed in favor of #14708

**Pull request closed without merging**